### PR TITLE
fix Confusion between _PIL_AVAILABLE and _IMAGE_AVAILABLE flags

### LINKED ISSUE #425: Confusion between _PIL_AVAILABLE and _IMAGE_AVAILABLE flags
## 🐛 Bug

In `flash.core.utilities.imports` the flag `_PIL_AVAILABLE` is defined but only used for a test (see: [link](https://github.com/PyTorchLightning/lightning-flash/search?q=_PIL_AVAILABLE))

For the import of PIL the `_IMAGE_AVAILABLE` flag is used, which is defined as `_IMAGE_AVAILABLE = _TORCHVISION_AVAILABLE and _TIMM_AVAILABLE and _KORNIA_AVAILABLE`.
The cases where `_IMAGE_AVAILABLE` is use to check for the import of PIL can be found [here](https://github.com/PyTorchLightning/lightning-flash/search?q=_IMAGE_AVAILABLE).

This causes errors for `datamodule.show_val_batch()`.

#### Code sample
```python
from flash.core.utilities.imports import _IMAGE_AVAILABLE, _module_available

print(_module_available("PIL")) # will print True if PIL is installed

print(_IMAGE_AVAILABLE) # will print False if torchvision, timm or kornia is missing
```

### Expected behavior

PIL should be imported when the _PIL_AVAILABLE is True and not the _IMAGE_AVAILABLE flag.

### PR DESCRIPTION
**Pull request by fstroth**
## What does this PR do?
Corrects the definition the `_IMAGE_AVAILABLE` flag and changed all import checks based on that flag, to check for each individual flag for a given import. 

Fixes #425 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/image/detection/test_data.py [hunk 1]
```diff
@@ -5,11 +5,10 @@
 import pytest
 
 from flash.core.data.data_source import DefaultDataKeys
-from flash.core.utilities.imports import _FIFTYONE_AVAILABLE, _IMAGE_AVAILABLE
+from flash.core.utilities.imports import _FIFTYONE_AVAILABLE, _PIL_AVAILABLE
 from flash.image.detection.data import ObjectDetectionData
-from tests.helpers.utils import _IMAGE_TESTING
```
Comment (ethanwharris) [inline tests/image/detection/test_data.py:10]: I think this line is needed for tests to pass

**Review by ethanwharris — COMMENTED** [review]
Awesome job!!! Some test failures but I think you just need to add back a line (see comment). Great work :smiley:

File: tests/image/detection/test_data.py (hunk 1 above)
Comment (fstroth) [reply tests/image/detection/test_data.py:10]: Didn't realize I removed the line. Thank you.

Comment (ethanwharris) [reply tests/image/detection/test_data.py:10]: No problem :smiley:

**Review by justusschock — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: flash/core/utilities/imports.py [hunk 2]
```diff
@@ -91,5 +91,6 @@ def _compare_version(package: str, op, version) -> bool:
 _TEXT_AVAILABLE = _TRANSFORMERS_AVAILABLE
 _TABULAR_AVAILABLE = _TABNET_AVAILABLE and _PANDAS_AVAILABLE
 _VIDEO_AVAILABLE = _PYTORCHVIDEO_AVAILABLE
-_IMAGE_AVAILABLE = _TORCHVISION_AVAILABLE and _TIMM_AVAILABLE and _KORNIA_AVAILABLE
+_IMAGE_AVAILABLE = _TORCHVISION_AVAILABLE and _TIMM_AVAILABLE and _PIL_AVAILABLE and _KORNIA_AVAILABLE and\
+    _MATPLOTLIB_AVAILABLE and _COCO_AVAILABLE and _FIFTYONE_AVAILABLE
```
Comment (Borda) [inline flash/core/utilities/imports.py:95]:
```suggestion
_IMAGE_AVAILABLE = (_TORCHVISION_AVAILABLE and _TIMM_AVAILABLE and _PIL_AVAILABLE and _KORNIA_AVAILABLE and
    _MATPLOTLIB_AVAILABLE and _COCO_AVAILABLE and _FIFTYONE_AVAILABLE)
```

**Pull request merged**